src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -95,5 +255,33 @@ public IObservable<string> GetPullRequestTemplate(ILocalRepositoryModel reposito
             return ret;
         }
 
+        void EnsurePullRefSpecExists(IRepository repo)
+        {
+            var spec = "+refs/pull/*/head:refs/remotes/origin/pr/*";
+            var origin = repo.Network.Remotes["origin"];
+            var existing = origin.FetchRefSpecs.FirstOrDefault(x => x.Specification == spec);
+
+            if (existing == null)
+            {
+                repo.Network.Remotes.Update(origin, x => x.FetchRefSpecs.Add(spec));
+            }
+        }
+
+        static string GetSafeBranchName(string name)
```

Comment:
Ah, I see. Ok, carry on. Is it possible for a PR title to only contain invalid characters? Then what would be the branch name? Just a single `-`?